text/000-systems-design.md

```diff
@@ -0,0 +1,272 @@
+<!-- - Feature Name: (fill me in with a unique ident, `my_new_feature`)
+- Start Date: (fill me in with today's date, 2020-12-11)
+- RFC PR: [flybywiresim/rfcs#0000](https://github.com/flybywiresim/rfcs/pull/0000)
+- A32NX Issue: [flybywiresim/a32nx#0000](https://github.com/flybywiresim/a32nx/issues/0000) -->
+
+# Summary
+[summary]: #summary
+
+Creation of the systems stalled. The code itself contains useful algorithms, but the (reasoning behind the) design is unclear. This document intends to help to achieve a good systems software design in whatever language is used. The document itself uses Rust for code examples.
+
+# Motivation
+[motivation]: #motivation
+
+Good software design makes implementing new features easier. While we cannot foresee every feature beforehand, going through the major features ensures a consistent design which is easy to work with and understand. In my opinion, good software design should primarily focus on defining the structural concepts that exist in the software. The amount of concepts should be  limited, as to not overburden those who develop within it with the continuous question of: "should I use concept x or y to do z?".
+
+# Requirements
+
+Before going through the requirements, some definitions:
+
+* **System**: Does not refer to a software system but to a part of the aircraft, i.e. electrical system.
+* **Model**: The part of the software that relates to actual parts of the aircraft, e.g. `ElectricalSystem`, `Engine`, `EngineGenerator`, etc.
+* **Software**: The thing we're building 🤡.
```

Comment:
Probably don't want to include the clown emoji since this is to be taken seriously.
```suggestion
* **Software**: The thing we're building.
```